### ISSUE TITLE
Add InferenceAggregator node

### ISSUE THREAD

**Issue by lalitpagaria** [obsei/obsei#163]
As mentioned in https://github.com/obsei/obsei/issues/153, let introduce InferenceAggregator in Obsei to aggregate Analyzer result on text chunks (Splits by TextSplitter) to compute final inference.

**Comment by lalitpagaria** [comment]
@akar5h fyi

**Comment by akar5h** [comment]
Thanks, @lalitpagaria . As per the discussion on  #153 
First thing to do is integrate the #160 TextSplitter changes with Analyzers , change the Analyzer to work with TextSplitter for texts with length greater than 512 .  Right now texts greater than max length are truncated ( see #113)

Then For InterfaceAggregator, I have two approaches in mind:
1. merge the output of the analyzer by simply averaging out the results of splits, I want to try out a simple count approach. The output label of text would be the label with the highest count among splits.
2. Another approach would be to combine the softmax logits outputs of N*K classes as mentioned in https://discuss.huggingface.co/t/new-pipeline-for-zero-shot-text-classification/681/84 discussion. This approach would need more clarity and thinking.  

So let's have two tasks in this Issue :
- [ ] Integrating TextSplitter and Analyzer
- [ ] InterfaceAggregator 
The first is something I've already started and then will continue on InterfaceAgrregator . Will share the results of both integration approaches so we can decide which one to move forward with. 
Kindly assign this to me

**Comment by lalitpagaria** [comment]
Yeah for this I already create this issue https://github.com/obsei/obsei/issues/164

> Integrating TextSplitter and Analyzer

**Comment by lalitpagaria** [comment]
@akar5h What do you feel about where to keep InterfaceAggregator?
I feel it is closely associated with Analyzer so better to keep them near them, but it will not be scalable solution in production multiple chucks of text can be processed by Analyzer running on different containers/instances. So InterfaceAggregator, nee to perform two tasks events aggregation and computing final Interface based on formula selected.

I am open for both type of solution -
1) Analyzer like ZeroShot, Sentiment etc will be subclass of InferenceAggregator. And similarly AnalyserConfig also will have InferenceAggregatorConfig. In this ways analyzer and aggregator perform in sequence.
2) Analyzer don't care about aggregator and simply analyze chunks and add raw output to meta. InferenceAggregator, later collect all chunks of doc_id and perform Aggregated Inference.

Please let me know which which one you feel good for first iteration? If you like we can setup a short call and discuss about this.

**Comment by akar5h** [comment]
I think the second approach is a more generic one. Yeah, we can still discuss it over a call.
